### PR TITLE
Transfer multiple nfts

### PR DESCRIPTION
**Pull request by rex-god**

### CODE REVIEW

File: src/modules/nft/components/transfer-nft/dto/transfer.dto.ts [hunk 1]
```diff
@@ -110,3 +110,44 @@ export class TransferNftDetachDto {
...
+    type: Array,
+    description: 'addresses of the NFT to be transfered',
+    example: ['HJ32KZye152eCFQYrKDcoyyq77dVDpa8SXE6v8T1HkBP'],
+  })
+  @IsNotEmpty()
+  @IsArray()
+  readonly token_addresses: string[];
+
+  @ApiProperty({
+    title: 'Address of wallet containing the NFT',
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/dto/transfer.dto.ts:135]: change it to `NFT owner's wallet address`

File: src/modules/nft/components/transfer-nft/dto/transfer.dto.ts [hunk 2]
```diff
@@ -110,3 +110,44 @@ export class TransferNftDetachDto {
...
+    type: String,
+    description: 'Address of the wallet containing the NFT to be transfered',
+    example: '2fmz8SuNVyxEP6QwKQs6LNaT2ATszySPEJdhUDesxktc',
+  })
+  @IsNotEmpty()
+  @IsString()
+  readonly from_address: string;
+
+  @ApiProperty({
+    title: 'To Wallet Address',
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/dto/transfer.dto.ts:145]: change it to `NFT receiver's wallet address`

File: src/modules/nft/components/transfer-nft/transfer-nft.controller.ts [hunk 3]
```diff
@@ -35,4 +35,16 @@ export class TransferNftController {
       result: res,
     };
   }
+
+  @Post('multiple_transfer')
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.controller.ts:39]: update the endpoint to `transfer_many`

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 4]
```diff
@@ -137,50 +115,101 @@ export class TransferNftService {
...
+        network,
+        token_addresses: tokenAddresses,
+        from_address: fromAddress,
+        to_address: toAddress,
+      } = transferNftDto;
+      const connection = Utility.connectRpc(network);
+      const fromAddressPubKey = new PublicKey(fromAddress);
+      const toAddressPubKey = new PublicKey(toAddress);
+
+      const txParts = chunk(tokenAddresses, 7);
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:161]: What is the size of transaction for submitting 7 NFTs?
And declare 7 as a constant at the top of this file.

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 5]
```diff
@@ -137,50 +115,101 @@ export class TransferNftService {
...
+        const serializedTransaction = tx.serialize({
+          requireAllSignatures: false,
+          verifySignatures: false,
+        });
+
+        const transactionBase64 = serializedTransaction.toString('base64');
+        txs.push(transactionBase64);
+      }
+
+      const nftReadEvent = new MultipleNftsWaitSyncEvent(network, toAddress, tokenAddresses, 25000);
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:207]: rename nftReadEvent to `nftsTransferedEvent`

File: src/modules/nft/components/transfer-nft/dto/transfer.dto.ts [hunk 6]
```diff
@@ -110,3 +110,44 @@ export class TransferNftDetachDto {
...
+    title: 'network',
+    type: String,
+    enum: WalletAdapterNetwork,
+    description: 'Select solana blockchain environment ',
+  })
+  @IsNotEmpty()
+  readonly network: WalletAdapterNetwork;
+
+  @ApiProperty({
+    title: 'Addresses of the NFT',
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/dto/transfer.dto.ts:125]: NFT Adresses

File: src/modules/nft/components/transfer-nft/dto/transfer.dto.ts [hunk 7]
```diff
@@ -110,3 +110,44 @@ export class TransferNftDetachDto {
...
+    example: ['HJ32KZye152eCFQYrKDcoyyq77dVDpa8SXE6v8T1HkBP'],
+  })
+  @IsNotEmpty()
+  @IsArray()
+  readonly token_addresses: string[];
+
+  @ApiProperty({
+    title: "NFT owner's wallet address",
+    type: String,
+    description: 'Address of the wallet containing the NFT to be transfered',
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/dto/transfer.dto.ts:137]: Address of the wallet containing the NFTs to be transferred

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts [hunk 8]
```diff
@@ -137,50 +117,101 @@ export class TransferNftService {
...
+        network,
+        token_addresses: tokenAddresses,
+        from_address: fromAddress,
+        to_address: toAddress,
+      } = transferNftDto;
+      const connection = Utility.connectRpc(network);
+      const fromAddressPubKey = new PublicKey(fromAddress);
+      const toAddressPubKey = new PublicKey(toAddress);
+
+      const txParts = chunk(tokenAddresses, maxTransferInstructionInSingleTx);
```
Comment (vgvishesh) [inline src/modules/nft/components/transfer-nft/transfer-nft.service.ts:163]: `maxNFTTransfersInOneTxn`

File: src/modules/nft/components/transfer-nft/dto/transfer.dto.ts (hunk 7 above)
Comment (rex-god) [reply src/modules/nft/components/transfer-nft/dto/transfer.dto.ts]: ✅

File: src/modules/nft/components/transfer-nft/transfer-nft.service.ts (hunk 8 above)
Comment (rex-god) [reply src/modules/nft/components/transfer-nft/transfer-nft.service.ts]: ✅

File: src/modules/nft/components/transfer-nft/dto/transfer.dto.ts (hunk 6 above)
Comment (rex-god) [reply src/modules/nft/components/transfer-nft/dto/transfer.dto.ts]: ✅

**Review by vgvishesh — APPROVED** [review]

**Pull request merged**